lib/index.js

```diff
@@ -255,9 +262,30 @@ function stream(arg, fn) {
   return s;
 }
 
+function combine(streams, fn) {
+  var i, s, deps, depEndStreams;
+  var endStream = createDependentStream([], trueFn);
+  deps = []; depEndStreams = [];
+  for (i = 0; i < streams.length; ++i) {
+    if (streams[i] !== undefined) {
+      deps.push(streams[i]);
+      if (streams[i].end !== undefined) depEndStreams.push(streams[i].end);
+    }
+  }
+  s = createDependentStream(deps, fn);
+  s.depsChanged = [];
+  s.fnArgs = s.deps.concat([s, s.depsChanged]);
+  s.end = endStream;
+  endStream.listeners.push(s);
+  addListeners(depEndStreams, endStream);
+  endStream.deps = depEndStreams;
+  updateStream(s);
+  return s;
+}
+
 var transduce = curryN(2, function(xform, source) {
   xform = xform(new StreamTransformer());
-  return stream([source], function(self) {
+  return combine([source], function(self) {
```

Comment:
Yeah, good catch.

When I address the changelist I posted I'll hit this as well as add more tests for the transduce code.